### PR TITLE
misc: final cleanup from audit results

### PR DESCRIPTION
**Pull request by mattverse**
Closes: #XXX

## What is the purpose of the change
This PR includes the final changes from audit results

## Testing and Verifying
Ensured that existing test cases pass
## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes?
  - [ ] Changelog entry added to `Unreleased` section of `CHANGELOG.md`?

Where is the change documented? 
  - [ ] Specification (`x/{module}/README.md`)
  - [ ] Osmosis documentation site
  - [ ] Code comments?
  - [ ] N/A

### CODE REVIEW

File: x/concentrated-liquidity/lp.go [hunk 1]
```diff
@@ -171,7 +171,7 @@ func (k Keeper) WithdrawPosition(ctx sdk.Context, owner sdk.AccAddress, position
 		return sdk.Int{}, sdk.Int{}, types.NotPositionOwnerError{PositionId: positionId, Address: owner.String()}
 	}
 
-	// Defense in depth, requestedLiquidityAmountToWithdraw should always be a positive value.
+	// Defense in depth, requestedLiquidityAmountToWithdraw should always be a value that is BGE than 0.
```
Comment (ValarDragon) [inline x/concentrated-liquidity/lp.go:174]:
```suggestion
	// Defense in depth, requestedLiquidityAmountToWithdraw should always be a value that is GE than 0.
```

(unless BGE is a term that I'm not familiar with)

File: x/concentrated-liquidity/model/pool.go [hunk 2]
```diff
@@ -226,10 +226,6 @@ func (p Pool) CalcActualAmounts(ctx sdk.Context, lowerTick, upperTick int64, liq
 		return sdk.Dec{}, sdk.Dec{}, types.ErrZeroLiquidity
 	}
 
-	if lowerTick >= upperTick {
```
Comment (ValarDragon) [inline x/concentrated-liquidity/model/pool.go:229]: why is this case impossible? Its in the public API so could be called externally, no?

Comment (mattverse) [reply x/concentrated-liquidity/model/pool.go:229]: `TicksToSqrtPrice` from the line below does the same check

File: x/concentrated-liquidity/lp.go [hunk 3]
```diff
@@ -194,23 +194,17 @@ func (k Keeper) WithdrawPosition(ctx sdk.Context, owner sdk.AccAddress, position
 		return sdk.Int{}, sdk.Int{}, err
 	}
 
-	// Retrieve the position in the pool for the provided owner and tick range.
-	positionLiquidity, err := k.GetPositionLiquidity(ctx, positionId)
-	if err != nil {
-		return sdk.Int{}, sdk.Int{}, err
+	// Check if the requested liquidity amount to withdraw is less than or equal to the available liquidity for the position.
+	// If it is greater than the available liquidity, return an error.
+	if requestedLiquidityAmountToWithdraw.GT(position.Liquidity) {
```
Comment (ValarDragon) [inline x/concentrated-liquidity/lp.go:199]: using position.liquidity directly is a good change! Looked above, and:
> positionHasActiveUnderlyingLockAndUpdate

does not modify the position (only a store for mapping), so I believe this is safe.

**Review by ValarDragon — APPROVED** [review]
LGTM, minor comments

File: x/concentrated-liquidity/model/pool.go (hunk 2 above)
Comment (ValarDragon) [reply x/concentrated-liquidity/model/pool.go:229]: I see that makes sense

**Comment by ValarDragon** [comment]
State breaking due to one less state read (and thus gas changes)

**Pull request merged**